oak_restricted_kernel_runtime/src/channel.rs

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use anyhow::anyhow;
+use oak_channel::{Read, Write};
+
+pub const OAK_CHANNEL_FD: i32 = 10;
+
+pub struct SyscallChannel {
```

Comment:
It's a channel implemented over a file descriptor (using read/write/fsync syscalls).

I agree that it can be confusing. I'll rename it to FileDescriptorChannel, as that's a more appropriate name.

The number 10 is completely arbitrary. I expect us to support two file descriptors in the future: one for the channel (say, 10) and stderr (standard value 2) for emergency logging.